src/components/Secrets/secret-modal.tsx

```diff
@@ -1,12 +1,15 @@
 import * as React from 'react';
-import { RemoteSecretModel } from '../../models';
-import { RemoteSecretKind } from '../../types';
+import { SecretModel } from '../../models';
+import { SecretKind, SecretType } from '../../types';
 import { createDeleteModalLauncher } from '../modal/DeleteResourceModal';
+import { unLinkSecretFromServiceAccount } from './utils/service-account-utils';
 
-export const secretDeleteModal = (secret: RemoteSecretKind) =>
+export const secretDeleteModal = (secret: SecretKind) =>
   createDeleteModalLauncher(secret.kind)({
     obj: secret,
-    model: RemoteSecretModel,
+    model: SecretModel,
+    submitCallback:
+      secret.type === SecretType.dockerconfigjson ? unLinkSecretFromServiceAccount : null,
```

Comment:
why use a conditional check here to pass the callback?

Reply:
linking/unlinking for imagePull Secret only, other types this callback not needed